src/transformers/models/textnet/modeling_textnet.py

```diff
@@ -0,0 +1,552 @@
+        hidden_states.append(hidden_state)
+
+        for block in self.stage2:
+            hidden_state = block(hidden_state)
+        hidden_states.append(hidden_state)
+
+        for block in self.stage3:
+            hidden_state = block(hidden_state)
+        hidden_states.append(hidden_state)
+
+        for block in self.stage4:
+            hidden_state = block(hidden_state)
+        hidden_states.append(hidden_state)
+
+        pooled_output = self.pooler(hidden_state)
+
+        if not return_dict:
+            output = (pooled_output, hidden_state)
+            return output + (hidden_states,) if output_hidden_states else output
+
+        return BaseModelOutputWithPoolingAndNoAttention(
+            pooler_output=pooled_output,
+            last_hidden_state=hidden_state,
+            hidden_states=tuple(hidden_states) if output_hidden_states else None,
+        )
+
+
+@add_start_docstrings(
+    """
+    TextNet backbone, to be used with frameworks like DETR and MaskFormer.
+    """,
+    TEXTNET_START_DOCSTRING,
+)
+class TextNetBackbone(TextNetPreTrainedModel, BackboneMixin):
+    def __init__(self, config):
+        super().__init__(config)
+        super()._init_backbone(config)
+
+        self.textnet = TextNetModel(config)
+        self.num_features = [
+            config.out_channels,
+            config.stage1_out_channels[-1],
+            config.stage2_out_channels[-1],
+            config.stage3_out_channels[-1],
+            config.stage4_out_channels[-1],
+        ]
+
+        # initialize weights and apply final processing
+        self.post_init()
+
+    @add_start_docstrings_to_model_forward(TEXTNET_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=BackboneOutput, config_class="")
+    def forward(
+        self, pixel_values: Tensor, output_hidden_states: Optional[bool] = None, return_dict: Optional[bool] = None
+    ) -> BackboneOutput:
+        """
+        Returns:
+
+        Examples:
+
+        ```python
+        # >>> from transformers import AutoImageProcessor, AutoBackbone
+        # >>> import torch
+        # >>> from PIL import Image
+        # >>> import requests
+        #
+        # >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        # >>> image = Image.open(requests.get(url, stream=True).raw)
+        #
+        # >>> processor = AutoImageProcessor.from_pretrained("google/resnetnv2-50")
+        # >>> model = AutoBackbone.from_pretrained("google/resnetnv2-50")
+        #
+        # >>> inputs = processor(image, return_tensors="pt")
+        # >>> outputs = model(**inputs)
+        ```"""
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+
+        outputs = self.textnet(pixel_values, output_hidden_states=True, return_dict=True)
```

Comment:
```suggestion
        outputs = self.textnet(pixel_values, output_hidden_states=True, return_dict=return_dict)
```
both tuple and dict outputs should be supported